src/lib.rs

```diff
@@ -385,7 +385,8 @@ pub trait Rng: RngCore {
     /// repeatedly with the same arguments, it will likely be faster to
     /// construct a [`Uniform`] distribution object and sample from that; this
     /// allows amortization of the computations that allow for perfect
-    /// uniformity within the [`Uniform::new`] constructor.
+    /// uniformity within the [`Uniform::new`] constructor and the
+    /// computations allow faster sampling with [`Uniform::sample`].
```

Comment:
Maybe like this instead?

> constructor, allowing faster sampling with [`Uniform::sample`] in some contexts.

(Important note: it will not always be faster.)

Also, I notice `Uniform::sample_single` above which doesn't exist any more (#454); can you fix?

Reply:
Oh right my bad. (Don't work on an old local version)

Reply:
> not always be faster

Are there contexts where the modulo calculations will produce a smaller `zone` than the leading zeros? Or is there some other reason for the difference?

Reply:
I think it's largely down to optimisation. `gen_range` sometimes optimises well with constants. But if the input is not known at compile time then `Uniform::new` will probably be faster.

Reply:
Perhaps 
> computations allow faster sampling with [`UniformInt::sample`] than [`UniformInt::sample_single`]?.

(I don't know about `Duration`.)

Reply:
I do like that mine mentions that it's the different calculations that mean better speed: implying that there are more factors for the speed than caching the computations.